lib/spack/spack/environment/environment.py

```diff
@@ -270,14 +272,14 @@ def read(name):
     return Environment(root(name))
 
 
-def create(name, init_file=None, with_view=None, keep_relative=False):
+def create(name, init_file=None, with_view=None, keep_relative=False, include_concrete=None):
```

Comment:
@becker33 With respect to what we were discussing yesterday, in #36927 this function:

https://github.com/spack/spack/blob/bd86ad12b3353f436224e4c427d0c1017d500405/lib/spack/spack/environment/environment.py#L275-L297

delegates to another one:

https://github.com/spack/spack/blob/bd86ad12b3353f436224e4c427d0c1017d500405/lib/spack/spack/environment/environment.py#L2464-L2521

to initialize the directory for the environment. When merging this feature PR and that refactor, I think any modification to get a consistent `spack.yaml` should be done in the `initialize_environment_dir` function I linked above.

Reply:
@alalazo Has this issue been addressed?

Reply:
Not really. This interface seems to be redundant. Why did we add a new argument `include_concrete`? Couldn't the presence of an included environment be inferred from the `init_file` ? Does it make sense to pass an `init_file` with included environments and not add `include_concrete` or vice-versa?